lib/src/bridge/device_connection_mixin.dart

```diff
@@ -14,13 +14,30 @@ mixin DeviceConnectionMixin on FlutterBLE {
       ArgumentName.requestMtu: requestMtu,
       ArgumentName.refreshGatt: refreshGatt,
       ArgumentName.timeoutMillis: timeout?.inMilliseconds
-    }).catchError((errorJson) =>
-            Future.error(BleError.fromJson(jsonDecode(errorJson.details))));
+    }).catchError(
+      (errorJson) => Future.error(
+        BleError.fromJson(jsonDecode(errorJson.details)),
+      ),
+    );
   }
 
   Stream<PeripheralConnectionState> observePeripheralConnectionState(
-      String identifier, bool emitCurrentValue) async* {
-    yield* _peripheralConnectionStateChanges
+      String identifier, bool emitCurrentValue) {
+    var controller = StreamController<String>(
+      onListen: () => _methodChannel
+          .invokeMethod(MethodName.observeConnectionState, <String, dynamic>{
+        ArgumentName.deviceIdentifier: identifier,
+        ArgumentName.emitCurrentValue: emitCurrentValue,
+      }).catchError(
```

Comment:
```suggestion
      },).catchError(
```
and then formatter :)